### PR TITLE
🐛 Source Instagram: Read previous state format and upgrade it

### LINKED ISSUE #4783: Instagram connector failing for all user-related tables
## Enviroment
- **Airbyte version**:  example is 0.27.0-alpha
- **OS Version / Instance**: GCP Compute Engine (n1-standard-1)
- **Deployment**: Docker
- **Source Connector and version**: Instagram 0.1.6
- **Destination Connector and version**: BigQuery 0.3.6
- **Severity**: High
- **Step where error happened**: Sync job 

## Current Behavior
I upgraded both Airbyte and the Instagram Connector on 9th July for housekeeping, however since that date the user_insights, user_lifetime_insights and users have been failing.

## Expected Behavior
Sync all tables as normal

## Logs
[logs-268-2.txt](https://github.com/airbytehq/airbyte/)
[logs-268-1.txt](https://github.com/airbytehq/airbyte/)
[logs-268-0.txt](https://github.com/airbytehq/airbyte/)

<details>
<summary>LOG</summary> 

```

replace this with
your long log
output here

```

</details>

## Steps to Reproduce
1. Set up Instagram Source
2. Set up BigQuery Destination
3. Sync (all defaults)

### PR DESCRIPTION
**Pull request by keu**
## What
Version 0.1.6 changed the format of state for the UserInsight stream.

## How
This PR adds support of the old format and auto-convert without losing values.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-instagram/source_instagram/streams.py [hunk 1]
```diff
@@ -356,8 +377,10 @@ def _get_insights(self, item, account_id) -> Optional[MutableMapping[str, Any]]:
             # An error might occur if the media was posted before the most recent time that
             # the user's account was converted to a business account from a personal account
             if error.api_error_subcode() == 2108006:
-                self.logger.error(f"Insights error for business_account_id {account_id}: {error.api_error_message()}")
-
+                details = error.body().get("error", {}).get("error_user_title") or error.api_error_message()
```
Comment (keu) [inline airbyte-integrations/connectors/source-instagram/source_instagram/streams.py:380]: this also improves error message

**Comment by davinchia** [comment]
@sherifnada would it be better to do this in a migration and not have such logic within connector code?

**Comment by keu** [comment]
@davinchia do we have a migration? Is this something related to the release of the Airbyte itself?
If so, I'm not sure we should release the new version of Airbyte every time some connector changes its state format.
The user is blocked by now.

**Comment by davinchia** [comment]
@keu yes we do have migrations for the main airbyte core. I guess neither way is great. This is fine for now. My concern is old code building up as more and more of these cases pop up.

**Review by davinchia — APPROVED** [review]
Approving to not block this.

**Comment by keu** [comment]
Thanks, @davinchia , I think the more cases we will have - the clearly we will see the solution.

**Pull request merged**